### PR TITLE
Parallax 1

### PR DESCRIPTION
**Pull request by alexodan**
# Description

First parallax challenge, the idea is to replicate the first portion of [this design](https://dribbble.com/shots/3979326-Area-2071-Landing-Page)

## Struggles

- Perspective (and `transform-origin`): Why the heck translateZ was messing with my elements positioning? Well, the aswer is intuitive (at least when you know about it). I was missing the fact objects look smaller/bigger depending on how further/closer from user's POV they are (basically Z axis). It also depends on where the object is located in terms of the X/Y axis and the point of origin for the main axis translation, since when centered the object won't translate (move) in that axis.

## Learnings

- Perspective (and `transform-origin`): Related with the struggle, it started to make some sense when taking into account the [vanishing point](https://www.artistic.photo/blog/the-vanishing-point) and how it impacted on the elements positioning and translations (Thanks Sandrina).

## References

- Perspective https://imjignesh.com/how-css-perspective-works/
- Parallax guide https://imjignesh.com/css-only-parallax/
- Parallax demo sites https://www.sliderrevolution.com/resources/css-parallax/
- Loom 😅

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,35 +1,15 @@
...
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
+    <div className="App">
```
Comment (sandrina-p) [inline src/App.tsx]: suggestion: Are you planning to do all the parallax exercises in this repo? If so, I highly recommend you changing this page (the root) to have only 3 links pointing to the page that has the actual parallax exercise. (you might need react-router (or [wouter](https://github.com/molefrog/wouter)) here :)

File: src/parallax/first/index.css [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  background-position: center center;
+}
+
+.p1-front-image {
+  width: 130px;
+  transform: translateZ(-100px) scale(1.33);
+  transform-origin: left;
+  position: absolute;
+  left: calc(41.6% + 233px);
+  top: calc(62% - 150px);
```
Comment (sandrina-p) [inline src/parallax/first/index.css]: question: explain me this, why position absolute with left and top? Have you noticed that this doesn't look very good (the same) in large screens vs medium screens? (the tricky with the zoom + and -

File: src/parallax/first/index.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import "./index.css";
```
Comment (sandrina-p) [inline src/parallax/first/index.tsx]: question: Are you aware that this makes the CSS global? You may want to consider using cssModules instead to not conflict with upcoming CSS classes

**Review by sandrina-p — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (alexodan) [reply src/App.tsx]: added a 'home' component in App that for now just displays a sad link (v0 😆

File: src/parallax/first/index.tsx (hunk 3 above)
Comment (alexodan) [reply src/parallax/first/index.tsx]: I added sass with modules, tho now I kinda wonder why I added sass at all but u know, might use it later?

File: src/parallax/first/index.css (hunk 2 above)
Comment (alexodan) [reply src/parallax/first/index.css]: completely refactored this 😅
just to mention how I came up with the grid numbers:
- I added a grid of 12 cols and 12 rows thinking I could place the font image doing something like `grid-area: 6 / 8 / 7 / 9;` to position it at the center of the right image. Turned out to be completely wrong haha cuz the grid cells dynamically resized based on the screen width/height.
- The 41.6 came from 100/12 = 8.66 * 5 (why 5 and not 6 which would be the middle cell grid? cuz I wanted the right img to be shifted a bit to the left)

funny enough when I tried zooming in/out I didn't see major changes in the layout so I completely missed that, regardless this ended up being a very wrong approach and conflicting a lot with perspective and translation in Z axis!

File: src/App.tsx (hunk 1 above)
Comment (sandrina-p) [reply src/App.tsx]: "make it work then make it better" so it's perfect for this PR :P

File: src/parallax/first/index.css (hunk 2 above)
Comment (sandrina-p) [reply src/parallax/first/index.css]: More important than knowing better solution it's understanding what was wrong with the initial solution. That's why I asked you to write it down (even after watching the looms I shared eheh). Here's a [happy mem](https://github.com/alexodan/parallax-demo/assets/14869087/1c1e8b87-c5a1-4f3b-8172-470203856382) :p

File: src/parallax/first/index.tsx (hunk 3 above)
Comment (sandrina-p) [reply src/parallax/first/index.tsx]: As long as it's scoped CSS anything works really ;) so don't overthink about it, it's all CSS at the end of the day.

**Review by sandrina-p — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,30 +1,26 @@
...
-- Install [eslint-plugin-react](https://github.com/jsx-eslint/eslint-plugin-react) and add `plugin:react/recommended` & `plugin:react/jsx-runtime` to the `extends` list
+## Learnings
+
+### First challenge
+
+- Perspective (and `transform-origin`): Related with the struggle, it started to make some sense when taking into account the [vanishing point](https://www.artistic.photo/blog/the-vanishing-point) and how it impacted on the elements positioning and translations (Thanks Sandrina).
+
+## References
+
+- Perspective https://imjignesh.com/how-css-perspective-works/
```
Comment (sandrina-p) [inline README.md:25]: oh wow this reference is amazing!

**Pull request merged**